File: fronted_dashbaord/club_dashboard/club.js
const API = {
  teams: '/api/club/teams',
  roster: '/api/club/roster',
  matches: '/api/club/matches',
  staff: '/api/club/staff',
  me: '/api/auth/me'
};

const $ = id => document.getElementById(id);

// TAB SWITCHING
document.querySelectorAll('.tab-btn').forEach(btn => {
  btn.addEventListener('click', () => {
    const tabName = btn.getAttribute('data-tab');
    switchTab(tabName);
  });
});

function switchTab(tabName) {
  // Hide all tabs
  document.querySelectorAll('.tab-content').forEach(tab => {
    tab.style.display = 'none';
  });
  
  // Remove active class
  document.querySelectorAll('.tab-btn').forEach(btn => {
    btn.classList.remove('active');
  });
  
  // Show selected tab
  document.getElementById(tabName + '-tab').style.display = 'block';
  
  // Add active class to button
  event.target.classList.add('active');
  
  // Load data for tab
  if (tabName === 'teams') loadTeams();
  else if (tabName === 'roster') loadRoster();
  else if (tabName === 'matches') loadMatches();
  else if (tabName === 'staff') loadStaff();
  else if (tabName === 'stats') loadStats();
}

// TEAMS
async function loadTeams() {
  try {
    const res = await fetch(API.teams, { credentials: 'include' });
    if (!res.ok) throw new Error(res.statusText);
    const teams = await res.json();
    renderTeams(teams);
  } catch (e) {
    alert('Failed to load teams: ' + e.message);
  }
}

function renderTeams(teams) {
  const tbody = document.querySelector('#teamsTable tbody');
  tbody.innerHTML = '';
  (teams || []).forEach(t => {
    const tr = document.createElement('tr');
    tr.innerHTML = `
      <td><strong>${escapeHtml(t.name)}</strong></td>
      <td>${escapeHtml(t.category || '—')}</td>
      <td>${escapeHtml(t.coach || '—')}</td>
      <td>${t.playerCount || 0}</td>
      <td>${t.founded || '—'}</td>
      <td><button class="btn" style="padding:6px 12px;font-size:0.8rem">Edit</button></td>
    `;
    tbody.appendChild(tr);
  });
}

function showAddTeamModal() {
  $('addTeamModal').style.display = 'block';
}

function closeModal(modalId) {
  $(modalId).style.display = 'none';
}

function saveTeam() {
  const name = $('teamName').value.trim();
  const category = $('teamCategory').value.trim();
  const coach = $('teamCoach').value.trim();
  
  if (!name || !category) {
    alert('Please fill in required fields');
    return;
  }
  
  const team = {
    name, category, coach,
    playerCount: 0,
    founded: new Date().getFullYear().toString()
  };
  
  fetch(API.teams, {
    method: 'POST',
    credentials: 'include',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(team)
  }).then(res => {
    if (res.ok) {
      closeModal('addTeamModal');
      loadTeams();
      alert('Team added successfully');
    }
  }).catch(e => alert('Error: ' + e.message));
}

// ROSTER
async function loadRoster() {
  try {
    const res = await fetch(API.roster, { credentials: 'include' });
    if (!res.ok) throw new Error(res.statusText);
    const roster = await res.json();
    renderRoster(roster);
  } catch (e) {
    alert('Failed to load roster: ' + e.message);
  }
}

function renderRoster(roster) {
  const tbody = document.querySelector('#rosterTable tbody');
  tbody.innerHTML = '';
  (roster || []).forEach(p => {
    const tr = document.createElement('tr');
    const statusColor = p.status === 'active' ? '#10b981' : '#ef4444';
    tr.innerHTML = `
      <td><strong>${escapeHtml(p.name)}</strong></td>
      <td>${escapeHtml(p.position || '—')}</td>
      <td>${p.number || '—'}</td>
      <td>${p.age || '—'}</td>
      <td><span style="background:${statusColor};color:#fff;padding:4px 8px;border-radius:4px;font-size:0.8rem">${p.status || 'active'}</span></td>
      <td>${p.joined || '—'}</td>
      <td><button class="btn" style="padding:6px 12px;font-size:0.8rem">Edit</button></td>
    `;
    tbody.appendChild(tr);
  });
}

// MATCHES
async function loadMatches() {
  try {
    const res = await fetch(API.matches, { credentials: 'include' });
    if (!res.ok) throw new Error(res.statusText);
    const matches = await res.json();
    renderMatches(matches);
  } catch (e) {
    alert('Failed to load matches: ' + e.message);
  }
}

function renderMatches(matches) {
  const tbody = document.querySelector('#matchesTable tbody');
  tbody.innerHTML = '';
  (matches || []).forEach(m => {
    const tr = document.createElement('tr');
    const statusColor = m.status === 'completed' ? '#6b7280' : m.status === 'ongoing' ? '#f59e0b' : '#3b82f6';
    tr.innerHTML = `
      <td>${m.date || '—'}</td>
      <td>${escapeHtml(m.homeTeam || '—')}</td>
      <td>${escapeHtml(m.awayTeam || '—')}</td>
      <td>${escapeHtml(m.venue || '—')}</td>
      <td><strong>${m.result || '—'}</strong></td>
      <td><span style="background:${statusColor};color:#fff;padding:4px 8px;border-radius:4px;font-size:0.8rem">${m.status || 'scheduled'}</span></td>
      <td><button class="btn" style="padding:6px 12px;font-size:0.8rem">Details</button></td>
    `;
    tbody.appendChild(tr);
  });
}

// STAFF
async function loadStaff() {
  try {
    const res = await fetch(API.staff, { credentials: 'include' });
    if (!res.ok) throw new Error(res.statusText);
    const staff = await res.json();
    renderStaff(staff);
  } catch (e) {
    alert('Failed to load staff: ' + e.message);
  }
}

function renderStaff(staff) {
  const tbody = document.querySelector('#staffTable tbody');
  tbody.innerHTML = '';
  (staff || []).forEach(s => {
    const tr = document.createElement('tr');
    tr.innerHTML = `
      <td><strong>${escapeHtml(s.name)}</strong></td>
      <td>${escapeHtml(s.position || '—')}</td>
      <td>${escapeHtml(s.department || '—')}</td>
      <td>${escapeHtml(s.contact || '—')}</td>
      <td>${s.since || '—'}</td>
      <td><button class="btn" style="padding:6px 12px;font-size:0.8rem">Edit</button></td>
    `;
    tbody.appendChild(tr);
  });
}

// STATS
function loadStats() {
  fetch(API.teams, { credentials: 'include' })
    .then(res => res.json())
    .then(teams => {
      $('totalTeams').textContent = teams.length;
      const totalPlayers = teams.reduce((sum, t) => sum + (t.playerCount || 0), 0);
      $('totalPlayers').textContent = totalPlayers;
    })
    .catch(e => console.error('Error loading stats:', e));
}

// BUTTON EVENTS
$('addTeamBtn').addEventListener('click', showAddTeamModal);
$('addPlayerRosterBtn').addEventListener('click', () => alert('Add player feature coming soon'));
$('addMatchBtn').addEventListener('click', () => alert('Schedule match feature coming soon'));
$('addStaffBtn').addEventListener('click', () => alert('Add staff feature coming soon'));

$('applyRosterBtn').addEventListener('click', () => {
  loadRoster();
});

$('clearRosterBtn').addEventListener('click', () => {
  $('teamFilter').value = '';
  $('playerSearchFilter').value = '';
  loadRoster();
});

function escapeHtml(s) {
  return String(s || '').replace(/[&<>"]/g, c => ({
    '&': '&amp;', '<': '&lt;', '>': '&gt;', '"': '&quot;'
  }[c]));
}

// Load initial data
document.addEventListener('DOMContentLoaded', () => {
  loadTeams();
});

File: backed_server/api/club.py
from flask import Blueprint, request, jsonify, session
import json
import os

bp = Blueprint('club_api', __name__)

DATA_DIR = 'backed_server/data_store'

def load_json(filename):
    filepath = os.path.join(DATA_DIR, filename)
    try:
        with open(filepath, 'r') as f:
            return json.load(f)
    except:
        return []

def save_json(filename, data):
    filepath = os.path.join(DATA_DIR, filename)
    os.makedirs(os.path.dirname(filepath), exist_ok=True)
    with open(filepath, 'w') as f:
        json.dump(data, f, indent=2)

# Check admin/manager role
def require_club_role(f):
    from functools import wraps
    @wraps(f)
    def decorated_function(*args, **kwargs):
        user = session.get('user')
        if not user or user.get('role') not in ['admin', 'manager']:
            return jsonify({'error': 'unauthorized'}), 403
        return f(*args, **kwargs)
    return decorated_function

# TEAMS
@bp.route('/api/club/teams', methods=['GET'])
def get_teams():
    teams = load_json('club_teams.json')
    return jsonify(teams), 200

@bp.route('/api/club/teams', methods=['POST'])
@require_club_role
def create_team():
    data = request.get_json() or {}
    teams = load_json('club_teams.json')
    
    team = {
        'id': str(len(teams) + 1),
        'name': data.get('name'),
        'category': data.get('category'),
        'coach': data.get('coach'),
        'playerCount': data.get('playerCount', 0),
        'founded': data.get('founded')
    }
    
    teams.append(team)
    save_json('club_teams.json', teams)
    return jsonify(team), 201

# ROSTER
@bp.route('/api/club/roster', methods=['GET'])
def get_roster():
    roster = load_json('club_roster.json')
    return jsonify(roster), 200

@bp.route('/api/club/roster', methods=['POST'])
@require_club_role
def create_roster_player():
    data = request.get_json() or {}
    roster = load_json('club_roster.json')
    
    player = {
        'id': str(len(roster) + 1),
        'name': data.get('name'),
        'position': data.get('position'),
        'number': data.get('number'),
        'age': data.get('age'),
        'status': data.get('status', 'active'),
        'joined': data.get('joined')
    }
    
    roster.append(player)
    save_json('club_roster.json', roster)
    return jsonify(player), 201

# MATCHES
@bp.route('/api/club/matches', methods=['GET'])
def get_matches():
    matches = load_json('club_matches.json')
    return jsonify(matches), 200

@bp.route('/api/club/matches', methods=['POST'])
@require_club_role
def create_match():
    data = request.get_json() or {}
    matches = load_json('club_matches.json')
    
    match = {
        'id': str(len(matches) + 1),
        'date': data.get('date'),
        'homeTeam': data.get('homeTeam'),
        'awayTeam': data.get('awayTeam'),
        'venue': data.get('venue'),
        'result': data.get('result'),
        'status': data.get('status', 'scheduled')
    }
    
    matches.append(match)
    save_json('club_matches.json', matches)
    return jsonify(match), 201

# STAFF
@bp.route('/api/club/staff', methods=['GET'])
def get_staff():
    staff = load_json('club_staff.json')
    return jsonify(staff), 200

@bp.route('/api/club/staff', methods=['POST'])
@require_club_role
def create_staff():
    data = request.get_json() or {}
    staff = load_json('club_staff.json')
    
    member = {
        'id': str(len(staff) + 1),
        'name': data.get('name'),
        'position': data.get('position'),
        'department': data.get('department'),
        'contact': data.get('contact'),
        'since': data.get('since')
    }
    
    staff.append(member)
    save_json('club_staff.json', staff)
    return jsonify(member), 201
